apache/apache.go

```diff
@@ -57,22 +48,11 @@ var (
 	}
 )
 
+//Apache plugin struct which gathers plugin specific data
 type Apache struct{}
 
-func getMetrics(webserver string, metrics []string) ([]plugin.MetricType, error) {
-	tr := &http.Transport{}
-	client := &http.Client{Transport: tr}
-	resp, err := client.Get(webserver)
-	if err != nil {
-		return nil, err
-	}
-	if resp.StatusCode != 200 {
-		defer resp.Body.Close()
-		return nil, errReqFailed
-	}
-	defer resp.Body.Close()
-
-	mtsmap := make(map[string]plugin.MetricType)
+func parseMetrics(resp *http.Response) (map[string][]string, error) {
+	mtsmap := make(map[string][]string)
```

Comment:
Do we need `parseMetrics` to return a `map[string][]string`. Line 64 and line 70 we add only 1 element to the slice, so we could return a `map[string]string`.